docker/dockerfile-generator.sh

```diff
@@ -412,12 +484,17 @@ processArgs "$@"
 generateFile
 generateConfig
 printPreamble
-printAptPackages
-# OpenJ9 MUST use gcc7, HS doesn't have to
-if [ ${OPENJ9} == true ]; then
-  printgcc
+if echo "${IMAGE}" | grep -i -e "fedora" -e "centos" -e "rocky" -e "stream" -e "rhel" ; then
+  printDnfPackages
+elif echo "${IMAGE}" | grep -i -e "ubuntu" -e "debian" ;  then
+  printAptPackages
+  # OpenJ9 MUST use gcc7, HS doesn't have to
+  if [ ${OPENJ9} == true ]; then
+    printgcc
```

Comment:
How does printgcc set it to 7?

Reply:
The diff is poorly formatted. See:
```
if echo "${IMAGE}" | grep -i -e "fedora" -e "centos" -e "rocky" -e "stream" -e "rhel" ; then
  printDnfPackages
elif echo "${IMAGE}" | grep -i -e "ubuntu" -e "debian" ;  then
  printAptPackages
  # OpenJ9 MUST use gcc7, HS doesn't have to
  if [ ${OPENJ9} == true ]; then
    printgcc
  fi
else
  echo "Unknown system, can not install build deps: $IMAGE"
fi
```

The `installAptPackages` installs gcc7 if openj9 is in play. the printgcc is making it used: `ENV CC=gcc-7 CXX=g++-7" >> "$DOCKERFILE_PATH". It is print, as it just prints the line to docker file.

I had not adapted the rpm part to openj9, as I never built it. If jdk needs different compiler, then in rhel/centos/rocky this is solved by scl gccX-devtools software collection, and its usage is different.  So am not going to do it any time  soon, unless there will be usecase/demand for it.  Then I will elaborate for sure.

Does it explain?